### PR TITLE
signal-desktop: Adds support for `aarch64-linux`

### PR DESCRIPTION
**Pull request by bkchr**
## Description of changes

This adds support for `signal-desktop` on `aarch64-linux`. This is achieved by leveraging a community deb package that builds it for `aarch64`. A future version may switches to a full custom build of the package, so that we don't need to use any debs at all.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/signal-desktop/default.nix:4]: Signal releases very frequently, we might have to make this two links in the future. It would be also nice to have an automated update script for the aarch64 deb archive as well as this breaks now my normal automation (nix-update).

Comment (Mic92) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix:4]: Also please add yourself as a maintainer as I cannot test the arm64 port.

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix
Comment (bkchr) [inline pkgs/applications/networking/instant-messengers/signal-desktop/default.nix:2]: @Mic92 is that what you had in mind?

Comment (Mic92) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix:2]: Almost we can keep the same attribute name `signal-desktop` but just have two separate version variables, one for aarch64 and one for the rest.

Comment (bkchr) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix:2]: So just one `version-aarch64` and one `version`? And then depending on the target architecture I choose which version to use? So basically what I have done before, but just two separate version variables?

Comment (Mic92) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix:2]: Yes.

Comment (bkchr) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: Is it now as you have imagined it @Mic92? If not, could you please give me an example.

**Review by Mic92 — APPROVED** [review]

**Comment by bkchr** [comment]
@Mic92 can you follow the order of the robo overloads and give this another approval? :D

**Comment by Mic92** [comment]
> @Mic92 can you follow the order of the robo overloads and give this another approval? :D

I was mainly waiting for eval to finish but left my approval in case someone else races me to merge.

**Comment by uninsane** [comment]
prior to this work (thank you!) i was building signal-desktop from-source in order to support aarch64. if there's interest more broadly in that, feel free to crib from this package: <https://github.com/nix-community/nur-combined/blob/83da3cf27ec66451b5edbb41ee11d164c2be5e86/repos/colinsane/pkgs/additional/signal-desktop-from-src/default.nix>

i never proposed a from-source PR here because i don't have the capacity to be a maintainer for this, and i haven't tested call functionality (yet). but i have used it for messaging daily over the last month and it's been stable. and i've updated Signal versions twice now, each time was painless. so if any maintainer is serious about a from-source build... message me or tag me on the PR for testing/debugging.

**Comment by bkchr** [comment]
Ahh nice work @uninsane. I also have some local nix file, but this is clearly not as advanced as your nix file:D

A source build would IMO be better as all platforms could use the same.

**Comment by teutat3s** [comment]
This leads to conflicts when backporting new versions, which happens frequently due to older versions stopping to work after 90 days. Let's try to backport this.

**Pull request merged**
